### PR TITLE
LAA: improve code in a couple of routines (NFC)

### PR DESCRIPTION
**Pull request by artagnon**

### CODE REVIEW

File: llvm/lib/Analysis/LoopAccessAnalysis.cpp [hunk 1]
```diff
@@ -1976,13 +1976,12 @@ MemoryDepChecker::getDependenceDistanceStrideAndSize(
...
   // invariant. We can generate a runtime check to disambiguate the accesses.
-  if (StrideAPtrInt == 0 || StrideBPtrInt == 0)
+  if (!StrideAPtrInt || !StrideBPtrInt)
     return MemoryDepChecker::Dependence::Unknown;
 
   // Both Src and Sink have a constant stride, check if they are in the same
   // direction.
-  if ((StrideAPtrInt > 0 && StrideBPtrInt < 0) ||
-      (StrideAPtrInt < 0 && StrideBPtrInt > 0)) {
+  if (StrideAPtrInt > 0 != StrideBPtrInt > 0) {
```
Comment (david-arm) [inline llvm/lib/Analysis/LoopAccessAnalysis.cpp:1984]: nit: Is it worth adding brackets around each check, i.e. `(StrideAPtrInt > 0) != (StrideBPtrInt > 0)`. Perhaps it's personal preference, but that reads a bit better.

**Review by david-arm — APPROVED** [review]
LGTM. I added one minor suggestion, but I don't feel too strongly about it.

File: llvm/lib/Analysis/LoopAccessAnalysis.cpp [hunk 2]
```diff
@@ -2040,19 +2039,16 @@ MemoryDepChecker::isDependent(const MemAccessInfo &A, unsigned AIdx,
...
 
-  const SCEVConstant *C = dyn_cast<SCEVConstant>(Dist);
+  const SCEVConstant *ConstDist = dyn_cast<SCEVConstant>(Dist);
 
   // Attempt to prove strided accesses independent.
-  if (C) {
-    const APInt &Val = C->getAPInt();
-    int64_t Distance = Val.getSExtValue();
+  if (ConstDist) {
+    int64_t Distance = std::abs(ConstDist->getAPInt().getSExtValue());
```
Comment (fhahn) [inline llvm/lib/Analysis/LoopAccessAnalysis.cpp:2046]: Can we use `ConstDist->getAPInt().abs()` here as is done below?

Comment (fhahn) [reply llvm/lib/Analysis/LoopAccessAnalysis.cpp:2046]: Also, if we take the absolute value, clearer to use uint64_t + getZExtValue?

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**